pkg/sources/adapter/googlecloudpubsubsource/adapter.go

```diff
@@ -124,7 +126,9 @@ func (a *adapter) handleMessage(ctx context.Context, msg *pubsub.Message) {
 
 	for _, event := range events {
 		if result := a.ceClient.Send(ctx, *event); !cloudevents.IsACK(result) {
-			sendErrs.errs = append(sendErrs.errs, err)
+			sendErrs.errs = append(sendErrs.errs,
+				fmt.Errorf("failed to send event with ID %s: %w", event.ID(), result),
+			)
 			continue
 		}
```

Comment:
This change isn't directly related to this PR, but I went ahead and fixed a bug because I saw some `"error": "[<nil>]"` in the logs of the receive adapter while troubleshooting E2E tests, which isn't particularly useful.